src/LineageTree/lineageTree.py

```diff
@@ -48,12 +45,198 @@ def get_next_id(self):
+    def add_branch(
+        self,
+        succ: int,
+        length: int,
+        move_timepoints: bool = True,
+        pos: [callable, None] = None,
+        reverse: bool = False,
+    ):
+        """Adds a branch of specific length to a node either as a successor or as a predecessor.
+        If it is placed on top of a tree all the nodes will move timepoints #length down.
+
+        Args:
+            succ (int): Id of the successor (predecessor if reverse is False)
+            length (int): The length of the new branch.
+            pos (np.ndarray, optional): The new position of the branch. Defaults to None.
+            move_timepoints (bool): Important only if reverse= True
+            reverese (bool): If reverse will add a successor branch instead of a predecessor branch
+        Returns:
+            (int): Id of the first node of the sublineage.
+        """
+        if length == 0:
+            return succ
+        if self.predecessor.get(succ, None) and not reverse:
+            raise Warning("Cannot add 2 predecessors to a node")
+        time = self.time[succ]
+        original = succ
+        if not reverse:
+            if move_timepoints:
+                nodes_to_move = set(self.get_sub_tree(succ))
+                new_times = {
+                    node: self.time[node] + length for node in nodes_to_move
+                }
+                for node in nodes_to_move:
+                    old_time = self.time[node]
+                    self.time_nodes[old_time].remove(node)
+                    self.time_nodes.setdefault(old_time + length, set()).add(
+                        node
+                    )
+                self.time.update(new_times)
+                for t in range(length - 1, -1, -1):
+                    _next = self.add_node(
+                        time + t,
+                        succ=succ,
+                        pos=self.pos[original],
+                        reverse=True,
+                    )
+                    succ = _next
+            else:
+                for t in range(length):
+                    _next = self.add_node(
+                        time - t,
+                        succ=succ,
+                        pos=self.pos[original],
+                        reverse=True,
+                    )
+                    succ = _next
+        else:
+            for t in range(length):
+                _next = self.add_node(
+                    time + t, succ=succ, pos=self.pos[original], reverse=False
+                )
+                succ = _next
+            self.labels[succ] = "New branch"
+        if self.time[succ] == 0:
+            self.roots.add(succ)
+            self.labels[succ] = "New branch"
+        if original in self.roots:
+            self.roots.add(succ)
+            self.labels[succ] = "New branch"
+            self.roots.remove(original)
+            self.labels.pop(original, -1)
```

Comment:
That is only true when `reverse` is `True` and/or `self.time[succ] == 0` right? I think you should change that.